app/frontend/src/components/Markdown.test.tsx

```diff
@@ -40,37 +47,68 @@ describe("Markdown widget", () => {
       </div>
     );
     expect(screen.queryByAltText("image")).toBeNull();
-    expect(screen.getByTestId("root")).toContainHTML(
-      '<a href="https://example.com">image</a>'
-    );
+    expect(
+      getByRole(screen.getByTestId("root"), "link", { name: "image" })
+    ).toHaveAttribute("href", "https://example.com");
   });
 
-  it("doesn't convert markdown image to link for allowed domain", () => {
+  it("doesn't convert markdown image to link for allowed domain for allowImages='couchers'", () => {
     const mediaURL = "https://mymedia.com";
     process.env.REACT_APP_MEDIA_BASE_URL = mediaURL;
     render(
-      <div>
-        <div data-testid="allowed">
-          <Markdown
-            source={`# MD\nan image: ![image](${mediaURL}/image.png)`}
-            topHeaderLevel={1}
-            allowImages="couchers"
-          />
-        </div>
-        <div data-testid="not-allowed">
-          <Markdown
-            source={`# MD\nan image: ![image](${mediaURL}/image.png)`}
-            topHeaderLevel={1}
-            allowImages="none"
-          />
-        </div>
+      <div data-testid="allowed">
+        <Markdown
+          source={`# MD\nan image: ![image](${mediaURL}/image.png)`}
+          topHeaderLevel={1}
+          allowImages="couchers"
+        />
       </div>
     );
-    expect(screen.getByTestId("allowed")).toContainHTML(
-      `<img src="${mediaURL}/image.png" alt="image">`
+    expect(
+      getByAltText(screen.getByTestId("allowed"), "image")
+    ).toHaveAttribute("src", "https://mymedia.com/image.png");
+    expect(
+      queryByRole(screen.getByTestId("allowed"), "link", { name: "image" })
+    ).toBeNull();
```

Comment:
```suggestion
    ).not.toBeInTheDocument();
```
Slightly more explicit with the `jest-dom` matchers. Same comment for all places here using `toBeNull`